ruby-on-rails/docs/gem-tutorials/spree.md

```diff
@@ -0,0 +1,324 @@
+
+##### Spree official documentation:
+
+- [The gem](https://github.com/spree/spree)
+- [For developers](https://guides.spreecommerce.org/developer/)
+- [For api](https://guides.spreecommerce.org/api/)
+- [Factories for testing](https://github.com/spree/spree/tree/master/core/lib/spree/testing_support/factories)
+
+##### Some useful spree extension:
+
+To manage social auth like Facebook, go [here](https://github.com/spree-contrib/spree_social).
+
+##### Some tutorials:
+
+- [Getting started with spree](https://www.youtube.com/watch?v=IgCVg-GaQ2k)
+- [Integrate stripe with spee](https://blog.botreetechnologies.com/integrate-stripe-payment-gateway-in-spree-commerce-71e32239ba27)
+- [Adding a custom model in spree database](https://nnodes.com/blog/2016/custom-model-with-spree)
+- [Create an extension for spree](https://nebulab.it/blog/developing-spree-extension-with-tdd/)
+
+##### Developers in Wolox with Spree experience
+
+- Gregoire Marnotte [Gmarn](https://github.com/Gmarn)
+
+
+
+## Basic Spree gem
+
+### First step tutorial
+
+On the official documentation of spree there is a perfect tutorial to start, [here](https://guides.spreecommerce.org/developer/getting_started_tutorial.html).
+
+We recommend to install spree gem with the version *3.4* because in this version you can handle frontend view easier.
+
+```ruby
+gem 'spree', '~> 3.4'
+gem 'spree_auth_devise', '~> 3.3'
+gem 'spree_gateway', '~> 3.3'
+```
+
+### A word on spree gem composition
+Spree is not an unic gem, in reality spree is composed of various gem, and when you bundle spree this gem will be install:
+
+- *spree*: base gem contain only the installer for other gems;
+- *spree_core*: this is the base which contains all models and testing support;
+- *spree_cmd*: spree command line utility to create extension;
+- *spree_api*: api of spree;
+- *spree_backend*: handle all the backoffice / admin;
+- *spree_frontend*: manage all views and front-end controller;
+
+The two other gems recently install, spree_auth_devise and spree_gateway, allow respectively to handle authentication with devise and user model and manage gateway for payment.
+
+## Some useful extensions
+
+There is to extension gem that we use in Enjoy-Vue:
+
+- spree_social: to manage authentication across facebook, twitter, github...
+```ruby
+gem 'spree_social', github: 'spree-contrib/spree_social'
+```
+- spree_mail_settings: to manage mail sending.
+```ruby
+gem 'spree_mail_settings', github: 'spree-contrib/spree_mail_settings'
+```
+
+
+## Database / Model tricks
+In some cases it could be useful to modify or create table in spree database. Spree allow that with a simple way but some things are good to have in mind.
+
+### Module spree for models
+
+Before to start there is an important fact to know, all model from spree are in the module **Spree**. This implies, when calling a model with the ORM, across the rails console for example, to cast **Spree::** before must be specify before.
```

Comment:
Before to start, there is an important fact to know, all the models from spree are in the module **Spree**. This implies *that*, when calling a model with the ORM, *using* the rails console for example, **Spree::** must be specified before your call.